api/include/opentelemetry/trace/trace_id.h

```diff
@@ -42,6 +42,9 @@ class TraceId final
     memcpy(rep_, id.data(), kSize);
   }
 
+  // Creates a TraceId with the given ID.
+  explicit TraceId(const uint8_t *data) noexcept { memcpy(rep_, data, kSize); }
```

Comment:
I'd prefer to stick with the safe `nostd::span`. It's not that much effort to convert a `const uint8_t *` to a `nostd::span<const uint8_t, kSize>`. Then we're on the safer side.

Reply:
Ok, it does not truly add much extra safety while initializing `TraceId` from random GUID fixed-size buffer of 16 bytes, for example. And adds an extra line of code to wrap the raw buffer into span to pass it. But I got  your point.